internal/stream/stream.go

```diff
@@ -184,9 +187,82 @@ func (s *Stream) StartReader(reader Reader) {
 
 	sr.start()
 
-	for _, sm := range s.streamMedias {
+	for m, sm := range s.streamMedias {
 		for _, sf := range sm.formats {
 			sf.startReader(sr)
+			if m.Type == description.MediaTypeVideo {
+				cb := sf.runningReaders[sr]
+				if cb == nil {
+					continue
+				}
+
+				framesWithAU := 0
+				for _, u := range s.CachedUnits {
+					if !isEmptyAU(u) {
+						framesWithAU++
+					}
+				}
+				if framesWithAU == 0 {
+					continue
+				}
+
+				// The previous p-frames must be sent at a certain speed to avoid the video freezing.
+				// We must update the PTS of the p-frames to have them played back real quick, but not instantly.
+				// If we do not update the PTS, the client will pause by an amount equal to the time between the p-frames.
+				// This is an issue because we want to send the p-frames as fast as possible.
+				playbackFPS := int64(100)
+				msPerFrame := 1000 / playbackFPS
+				ticksPerMs := int64(90000 / 1000)
+				pts := s.CachedUnits[len(s.CachedUnits)-1].GetPTS() - ticksPerMs*int64(framesWithAU)*msPerFrame
+				start := time.Now()
+				for _, u := range s.CachedUnits {
+					if isEmptyAU(u) {
+						continue
+					}
+					pts += ticksPerMs * msPerFrame
+					start = start.Add(time.Millisecond * time.Duration(msPerFrame))
+
+					var clonedU unit.Unit
+					switch tunit := u.(type) {
+					case *unit.H264:
+						clonedU = &unit.H264{
+							Base: unit.Base{
+								RTPPackets: []*rtp.Packet{
+									{
+										Header: rtp.Header{
+											Timestamp: uint32(pts),
+										},
+									},
+								},
+								PTS: pts,
+							},
+							AU: tunit.AU,
+						}
+					case *unit.H265:
+						clonedU = &unit.H265{
+							Base: unit.Base{
+								RTPPackets: []*rtp.Packet{
+									{
+										Header: rtp.Header{
+											Timestamp: uint32(pts),
+										},
+									},
+								},
+								PTS: pts,
+							},
+							AU: tunit.AU,
+						}
+					}
```

Comment:
Nit: Could extract this into a function for readability.

Reply:
It's a different type though!